guardian/itunes-rss
# Implement CDK

## What does this change?

- Switches the build system to Github Action
- Deprecates cloudformation.yaml in favour of CDK
- Updates to DEB artifact
- Ties it all together with Riffraff
- Reads CAPI key from Secrets Store (planning for auto-rotation)
- Reads Play! secrets from SSM

## How to test

This is now live on PROD, with 50% traffic (at the time of writing).

If you run `curl -D- http://itunes-feed.content-aws.guardianapis.com/culture/series/university-of-cambridge-festival-of-ideas/podcast.xml`, you should get a podcast RSS.  Keep this up enough times and it should have been served by this update.
If you run `curl -D- https://itunes-feed.content-aws.guardianapis.com/culture/series/university-of-cambridge-festival-of-ideas/podcast.xml`, you should also get a podcast RSS which is always served by this update. (Route53 is smart enough to not route https to the LB that is not configured for it).

## How can we measure success?

Modern deployment standards

## Have we considered potential risks?

We don't want to risk taking down the RSS feeds rolling this out. For this reason, we will run dual-stack and do a phased switchover gradually putting more traffic onto the new setup.

IAM policies have been copied over verbatim from cloudformation.yml but several seem over-broad. We'll leave it for a future PR to tighten these up.
